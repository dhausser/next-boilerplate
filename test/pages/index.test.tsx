import React from 'react'
// import { render, fireEvent } from '../testUtils'
import { render, fireEvent, waitFor } from '@testing-library/react'
import Home from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    waitFor(() => expect(asFragment()).toMatchSnapshot())
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})

    waitFor(() => {
      window.alert = jest.fn()
      fireEvent.click(getByText('Test Button'))
      expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
    })
  })
})
