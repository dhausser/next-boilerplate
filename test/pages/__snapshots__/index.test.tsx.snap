// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Failed to load
  </div>
</DocumentFragment>
`;

exports[`Home page matches snapshot 2`] = `<DocumentFragment />`;
